src/web_app_skeleton/docker/dev_entrypoint.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# This is the entrypoint script used for development docker workflows. By
+# default it will:
+#  - Install shards.
+#  - Run migrations.
+#  - Start the dev server.
+# It also accepts any commands to be run instead.
+
+
+warnfail () {
+  echo "$@" >&2
+  exit 1
+}
+
+case ${1:-} in
+  "") # If no arguments are provided, start lucky dev server.
+    ;;
+
+  *) # If any arguments are provided, execute them instead.
+    exec "$@"
+esac
+
+echo "Installing npm packages..."
+npm install
+
+if ! shards check ; then
+  echo "Installing shards..."
+  shards install
+fi
+
```

Comment:
I played around with running migrations automatically, but ultimately decided it wasn't worth it. There are a bunch of reasons why I can think that would be a bad idea, and it _really_ slows down startup because migrations require compilation before they can even be checked.

Reply:
Line 38 could maybe be:
`lucky dev || lucky db.migrate || warnfail "Lucky dev server failed."`

Reply:
That could produce some weird experiences. If lucky dev runs for 20 min and then fails because of a crystal compiler bug, the app would try to run migrations.

Reply:
This isn't hosting, this is the dev entrypoint

Reply:
Prebuilding the migrations in a release container is a great idea -- I do this in several apps and it's very handy.